spatialmath/box.go

```diff
@@ -54,9 +54,12 @@ func NewBox(pose Pose, dims r3.Vector, label string) (Geometry, error) {
 	return &box{pose, [3]float64{0.5 * dims.X, 0.5 * dims.Y, 0.5 * dims.Z}, label}, nil
 }
 
-// Label returns the label of the box.
+// Label returns the label of this box.
 func (b *box) Label() string {
-	return b.label
+	if b != nil {
```

Comment:
good catch! It still slips my mind that `nil` essentially can be typed in Go ...

Reply:
Yeah, I can see issues like this causing bugs down the line. I always forget. One day I'll find a good static analyzer to integrate into our workflow that catches stuff like this for us.